proto/secret/v1/secret.proto

```diff
@@ -0,0 +1,58 @@
+syntax = "proto3";
+package nitric.secret.v1;
+
+//protoc plugin options for code generation
+option go_package = "nitric/v1;v1";
+option java_package = "io.nitric.proto.secret.v1";
+option java_multiple_files = true;
+option java_outer_classname = "Secrets";
+option php_namespace = "Nitric\\Proto\\Secret\\V1";
+option csharp_namespace = "Nitric.Proto.Secret.v1";
+
+// The Nitric Secret Service contract
+service SecretService {
+  //Updates a secret, creating a new one if it doesn't already exist
+  rpc Put (SecretPutRequest) returns (SecretPutResponse);
```

Comment:
I would probably recommend using the ```Set``` operation name. 

```rpc Set (SecretSetRequest) returns (SecretSetResponse);```

I did this for the document service API to avoid confusion around the Dynamodb Put and Update operations which have different behaviour, and make these APIs more similar.

Otherwise contract API looks good.

Reply:
Normally I would agree, but in this case the Put is always an append operation (creates a new secret version with each Put). Where I think Set implies the ability to update/override an existing value rather than creating.

Reply:
Make sense, I presume the developer needs to manage the version.

Reply:
Correct, it's returned each time a secret is created. However you can also use 'latest' as an alias to return the latest version of a given secret.